### PR TITLE
feat: encrypt & decrypt payloads using jwe

### PR DESCRIPTION
**Pull request by Zangetsu101**

### CODE REVIEW

File: packages/dci-api/src/crypto/encrypt.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+const jwksSchema = z.object({
+  keys: z.array(jwkSchema)
+})
+
+export async function encryptPayload(
+  jwksUrl: string,
+  payload: components['schemas']['SearchResponse']
+): Promise<components['schemas']['EncryptedMessage']> {
+  const url = new URL(jwksUrl)
+  const response = await fetch(url)
```
Comment (naftis) [inline packages/dci-api/src/crypto/encrypt.ts]: Would it make sense to cache this to not need to refetch the jwks every time? Guess we could have something like:

```
function fetchJwks(url: string) {
  const url = new URL(jwksUrl)
  const response = await fetch(url)
  ...
}

const memoizedFetchJwks = memoize(fetchJwks) // https://lodash.com/docs/4.17.15#memoize
```
Or would it cause an issue if the JWKS gets updated

File: packages/dci-api/src/crypto/encrypt.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  }
+  if (jwe.header.enc === undefined) {
+    throw new Error('Missing required header paramater "enc" after encrypting')
+  }
+  if (jwe.header.kid === undefined) {
+    throw new Error('Missing required header paramater "kid" after encrypting')
+  }
+  if (jwe.encrypted_key === undefined) {
+    throw new Error('encrypted_key is required after encrypting')
+  }
```
Comment (naftis) [inline packages/dci-api/src/crypto/encrypt.ts]: In which cases would these be missing? Can it happen

**Review by naftis — APPROVED** [review]

File: packages/dci-api/src/crypto/encrypt.ts (hunk 1 above)
Comment (Zangetsu101) [reply packages/dci-api/src/crypto/encrypt.ts]: yea ideally we could have some kind of a cache timeout like here: https://github.com/panva/jose/blob/358e864a0cccf1e0f9928a959f91f18f3f06a7de/docs/interfaces/jwks_remote.RemoteJWKSetOptions.md. Unfortunately, this is configured to be used with "signing" keys and not "encrytion" ones. So we would need to implement it ourselves I guess

File: packages/dci-api/src/crypto/encrypt.ts (hunk 2 above)
Comment (Zangetsu101) [reply packages/dci-api/src/crypto/encrypt.ts]: They shouldn't be, I was mainly doing this to make typescript happy. :sweat_smile:

**Pull request merged**